app/jobs/file_set_index_job.rb

```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+class FileSetIndexJob < Hyrax::ApplicationJob
+  def perform(file_set)
+    file_set&.update_index
```

Comment:
this might need to become
```suggestion
    Hyrax.index_adapter.save(resource: file_set)
```